File: Chat-Example/src/main/java/client/protocols/Print.java
package client.protocols;

import client.ConnectionToChatServerHandler;
import com.google.gson.JsonObject;
import hoten.serving.message.JsonMessageHandler;

public class Print extends JsonMessageHandler<ConnectionToChatServerHandler> {

    @Override
    protected void handle(ConnectionToChatServerHandler connection, JsonObject data) {
        String msg = data.get("msg").getAsString();
        connection.getChat().announce(msg);
    }
}

File: Chat-Example/src/main/csharp/ChatClient/ChatClient/Protocol/Print.cs
using Newtonsoft.Json.Linq;
using Serving;
using System;

namespace ChatClient.Protocol
{
    class Print : JsonMessageHandler<ConnectionToChatServerHandler>
    {
        protected override void Handle(ConnectionToChatServerHandler connection, JObject data)
        {
            var msg = data["msg"].Value<String>();
            connection.Chat.Announce(msg);
        }
    }
}
